code/ensemble/utils/use-ensemble-criteria.R

```diff
@@ -0,0 +1,60 @@
+# Load models according to inclusion criteria before ensembling
+# 
+# Steps:
+# Currently, models included based on having: 
+# 1. All quantiles
+# 2. 4 horizons 
+# 3. Not manually specified for exclusion 
+# 4. Not the hub ensemble
+# 
+# Returns a list:
+# - "forecasts" = tibble, all forecasts passing inclusion criteria
+# - "criteria" = tibble, model names and criteria assessment
+# 
+library(dplyr)
+library(tibble)
+
+use_ensemble_criteria <- function(forecasts = 
+                                    covidHubUtils::load_forecasts(source = "local_hub_repo",
+                                                                  hub_repo_path = here(),
+                                                                  hub = "ECDC",
+                                                                  forecast_dates = forecast_date),
+                                  exclude_models = NULL,
+                                  team_name = "EuroCOVIDhub",
+                                  forecast_date = floor_date(today(), "week", 1)) {
+  
+  # 1. Identify models with all quantiles
+  quantiles <- round(c(0.01, 0.025, seq(0.05, 0.95, by = 0.05), 0.975, 0.99), 3)
+  
+  all_quantiles <- forecasts %>%
+    group_by(model, target_variable, location, target_end_date) %>%
+    summarise(n_quantiles = n(), 
+              .groups = "drop") %>%
+    filter(n_quantiles != length(quantile)) %>%
+    pull(model) %>%
+    unique()
+  
+  # 2. Identify models with 4 week forecasts
+  all_horizons <- forecasts %>%
+    group_by(model, target_variable, location) %>%
+    filter(any(grepl("4", horizon))) %>%
+    pull(model) %>%
+    unique()
+  
+  criteria <- tibble("forecast_date" = forecast_date, 
```

Comment:
If you keep the data frame and don't pull the model (as I suggest above) I think you can then just do 

```{r}
criteria = all_quantiles %>%
  inner_join(all_horizons, by = c("model", "target_variable", "location") %>%
  [your code from line 49 onwards]
```